service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderController.java

```diff
@@ -143,7 +143,7 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
             order = caseManagementOrderService.addNextHearingToCMO(caseData.getNextHearingDateList(), order);
         }
 
-        Document document = getDocument(caseData, order.isDraft());
+        Document document = getDocument(caseData.toBuilder().caseManagementOrder(order).build(), order.isDraft());
```

Comment:
could we pass `caseData.toBuilder().caseManagementOrder(order).build()` into a different variable?

Reply:
no need to, it's not used again. I think we can definitely clean up how this works in a separate tech debt. I already have one open to remove all this logic from the controller

Reply:
Cool. Looked at it and there was a lot happening. But I agree it can be cleaned up in a TD.